src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -1009,6 +1012,64 @@ private static bool TryReplaceWithAlias(this ExpressionSyntax node, SemanticMode
                 return false;
             }
 
+            // Avoid the TryReplaceWithAlias algorithm if the tree has no using alias directives. Since the input node
+            // might be a speculative node (not fully rooted in a tree), we use the original semantic model to find the
+            // equivalent node in the original tree, and from there determine if the tree has any using alias
+            // directives.
+            var originalModel = semanticModel;
+            while (originalModel.IsSpeculativeSemanticModel)
```

Comment:
I don't think you can have nested ones? We have SemanticModelExtensions.GetOriginalSemanticModel() that we use in many places that implies that's not a concern. In any case, might want to use the helper here?

Reply:
➡️ Used `GetOriginalSemanticModel`.